pkg/sdk/testint/secrets_gen_integration_test.go

```diff
@@ -0,0 +1,722 @@
+
+		details, err := client.Secrets.Describe(ctx, id)
+		require.NoError(t, err)
+
+		assert.NotContains(t, details.OauthScopes, "foo")
+		assert.NotContains(t, details.OauthScopes, "bar")
+		assert.Empty(t, details.OauthScopes)
+	})
+
+	t.Run("Create: SecretWithOAuthAuthorizationCodeFlow - refreshTokenExpiry date format", func(t *testing.T) {
+		id := testClientHelper().Ids.RandomSchemaObjectIdentifier()
+		request := sdk.NewCreateWithOAuthAuthorizationCodeFlowSecretRequest(id, "foo", refreshTokenExpiryTime, integrationId).
+			WithComment("a").
+			WithIfNotExists(true)
+
+		err := client.Secrets.CreateWithOAuthAuthorizationCodeFlow(ctx, request)
+		require.NoError(t, err)
+		t.Cleanup(testClientHelper().Secret.DropFunc(t, id))
+
+		_, err = client.Secrets.ShowByID(ctx, id)
+		require.NoError(t, err)
+
+		assertions.AssertThat(t,
+			objectassert.Secret(t, id).
+				HasName(id.Name()).
+				HasComment("a").
+				HasSecretType("OAUTH2").
+				HasDatabaseName(id.DatabaseName()).
+				HasSchemaName(id.SchemaName()),
+		)
+
+		details, err := client.Secrets.Describe(ctx, id)
+		require.NoError(t, err)
+
+		assertSecretDetails(details, secretDetails{
+			Name:                        id.Name(),
+			SecretType:                  "OAUTH2",
+			Comment:                     sdk.String("a"),
+			OauthRefreshTokenExpiryTime: stringDateToSnowflakeTimeFormat(time.DateOnly, refreshTokenExpiryTime),
+			IntegrationName:             sdk.String(integrationId.Name()),
+		})
+	})
+
+	t.Run("Create: SecretWithOAuthAuthorizationCodeFlow - refreshTokenExpiry datetime format", func(t *testing.T) {
+		refreshTokenWithTime := refreshTokenExpiryTime + " 12:00:00"
+		id := testClientHelper().Ids.RandomSchemaObjectIdentifier()
+
+		request := sdk.NewCreateWithOAuthAuthorizationCodeFlowSecretRequest(id, "foo", refreshTokenWithTime, integrationId)
+
+		err := client.Secrets.CreateWithOAuthAuthorizationCodeFlow(ctx, request)
+		require.NoError(t, err)
+		t.Cleanup(testClientHelper().Secret.DropFunc(t, id))
+
+		details, err := client.Secrets.Describe(ctx, id)
+		require.NoError(t, err)
+
+		assertSecretDetails(details, secretDetails{
+			Name:                        id.Name(),
+			SecretType:                  "OAUTH2",
+			OauthRefreshTokenExpiryTime: stringDateToSnowflakeTimeFormat(time.DateTime, refreshTokenWithTime),
+			IntegrationName:             sdk.String(integrationId.Name()),
+		})
+	})
+
+	t.Run("Create: WithBasicAuthentication", func(t *testing.T) {
+		comment := random.Comment()
+		id := testClientHelper().Ids.RandomSchemaObjectIdentifier()
+		request := sdk.NewCreateWithBasicAuthenticationSecretRequest(id, "foo", "foo").
+			WithComment(comment).
+			WithIfNotExists(true)
+
+		err := client.Secrets.CreateWithBasicAuthentication(ctx, request)
+		require.NoError(t, err)
+		t.Cleanup(testClientHelper().Secret.DropFunc(t, id))
+
+		_, err = client.Secrets.ShowByID(ctx, id)
+		require.NoError(t, err)
+
+		assertions.AssertThat(t,
+			objectassert.Secret(t, id).
```

Comment:
Just so you know, If you want to fetch the object earlier like in this case you can use `objectassert.SecretFromObject` and pass the prefetched object there to reduce unnecessary calls to Snowflake.